src/frontend/magnify/src/components/auth/SignupForm/SignupForm.tsx

```diff
@@ -125,15 +126,15 @@ export default function SignupForm() {
         )
         .required(),
     });
-  }, []);
+  }, [locale.currentLocale]);
 
   const initialValues: SignupFormValues = {
     name: '',
     email: '',
     username: '',
     password: '',
     confirmPassword: '',
-    language: MagnifyLocales.EN,
+    language: locale.currentLocale ?? MagnifyLocales.EN,
```

Comment:
Hmm.. `currentLocale` cannot be `null` or `undefined` as TranslationProvider defaults to `MagnifyLocales.EN`

Reply:
of course